tests/test_errors.py

```diff
@@ -580,7 +581,21 @@ def finalize(self, cfg: Any) -> None:
     pytest.param(
         Expected(
             create=lambda: None,
-            op=lambda cfg: OmegaConf.structured(IllegalType()),
+            op=lambda _: OmegaConf.structured(
+                ConcretePlugin(params=ConcretePlugin.FoobarParams(foo="x"))  # type: ignore
+            ),
+            exception_type=ValidationError,
+            msg="Value 'x' could not be converted to Integer",
+            object_type=ConcretePlugin.FoobarParams,
+            key="foo",
+            parent_node=lambda _: {},  # dummy parent
+        ),
+        id="structured:create_with_invalid_value",
```

Comment:
I've added the test case that you mentioned.
If I write `full_key="params.foo",` this results in `AssertionError: assert 'foo' == 'params.foo'`.
I think this is related to the use of a "dummy parent" in the `get_dataclass_data`/`get_attr_data` methods.